wordpress-mobile/WordPress-Android
# [Unit Tests] Remove mockito-core dependency

## Issue
We are using both `mockito-core` and `mockito-kotlin` dependencies in our codebase, but the latter transitively depends on the former. We have a issue though that `mockito-kotlin` relies on a different `mockito-core` version than what's declared in the project (which was recently updated to 5.10.0 https://github.com/wordpress-mobile/WordPress-Android/pull/20246).

After that version bump mentioned above, gradle and Android Studio started complaining, on all `@Mock` annotations, that it "Cannot access class 'org.mockit.Answers'." due to missing and conflicting dependencies. Full error message:

![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/2e6b693f-dbf1-4176-b391-b6258e6fb928)

## Solution
Because of that conflict, we should be using only `mockito-kotlin`, which introduces the Kotlin-specific functionality while still transitively bringing in the `mockito-core` lib, so we can use it in Java as well.

This PR addresses that by removing `mockito-core` from our dependencies. There are 2 important points to be noted though, see below.

### 1. Mockito 5.X min Java version
[Mockito 5.0.0 changed the minimum JVM target support to Java 11](https://github.com/mockito/mockito/releases/tag/v5.0.0) so, in theory (and as long as I'm not missing anything) it shouldn't really work for us, [since our `jvmTarget` is Java 8](https://github.com/wordpress-mobile/WordPress-Android/blob/345f54d21d835f24ea2619e860c794fd6f3ee842/build.gradle#L177), including test runs.

Our project has been "using" `mockito-core` 5.X for some time now though without running into issues, so my guess is that since most of our tests run in Kotlin, we were actually using stuff from 4.X this whole time, since `mockito-kotlin` is on version 4.1.0. 

I tried updating `mockito-kotlin` to the latest available version (5.2.1) and then I started getting compilation errors regarding inline bytecode using a different version (11) than our project (8), which matches the [notes from Mockito 5.0.0](https://github.com/mockito/mockito/releases/tag/v5.0.0), so I'm keeping 4.1.0 there, which is the latest `mockito-kotlin` 4 version available.

### 2. `mockito-android`
I kept `mockito-android` in our instrumented tests dependencies because otherwise those tests don't run properly. But to keep things consistent I downgraded it to 4.5.1, so it also depends on `mockito-core:4.5.1`, which is the version used by `mockito-kotlin:4.1.0` as well.

-----

## To Test:

Make sure unit tests run successfully for WordPress Vanilla Release and instrumented tests still pass successfully for both Jetpack Vanilla Debug and Wordpress Vanilla Debug, which are the suites that run on our CI currentlly.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Tests breakage

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Ran unit and instrumented tests and check CI passes

3. What automated tests I added (or what prevented me from doing so)

    - Updated any tests that required updating to properly use `mockito-kotlin`
